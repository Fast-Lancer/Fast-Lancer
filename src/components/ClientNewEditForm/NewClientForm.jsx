import { useState } from 'react'
import { useUser } from '../../context/UserContext'

export default function NewClientForm({ formSubmit }) {
  const { user } = useUser()
  const [client_name, setClient_name] = useState('')
  const [email, setEmail] = useState('')
  const [phone, setPhone] = useState('')
  const [business_name, setBusiness_name] = useState('')
  const [notes, setNotes] = useState('')

  const handleSubmit = async (e) => {
    e.preventDefault()
    try {
      await formSubmit({ client_name, email, phone, business_name, notes, user_id: user.id })
    } catch (err) {
      'err'
    }
  }

  return <div>
    <h2>Client Details</h2>
    <fieldset>
      <form onSubmit={handleSubmit}>
        <label htmlFor='client_name'>Client Name</label>
        <input
          id='client_name'
          type='text'
          value={client_name}
          required
          onChange={(e) => setClient_name(e.target.value)} />

        <label htmlFor='email'>E-Mail</label>
        <input
          id='email'
          type='email'
          value={email}
          required
          onChange={(e) => setEmail(e.target.value)} />

        <label htmlFor='phone'>Phone Number</label>
        <input
          id='phone'
          type='number'
          value={phone}
          required
          onChange={(e) => setPhone(e.target.value)} />

        <label htmlFor='business_name'>Business Name</label>
        <input
          id='business_name'
          type='text'
          value={business_name}
          placeholder='Optional'
          onChange={(e) => setBusiness_name(e.target.value)} />

        <label htmlFor='notes'>Notes</label>
        <textarea
          id='notes'
          placeholder='Optional'
          value={notes}
          onChange={(e) => setNotes(e.target.value)}
        />
        <button type='submit'>Save</button>
      </form>
    </fieldset>
  </div>
}
