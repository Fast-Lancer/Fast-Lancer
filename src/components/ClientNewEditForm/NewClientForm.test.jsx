import { render } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import NewClientForm from './NewClientForm'

it('renders the Home view accurately', () => {
  const { container } = render(
    <Router>
      <NewClientForm />
    </Router>
  )
  expect(container).toMatchSnapshot()
})
