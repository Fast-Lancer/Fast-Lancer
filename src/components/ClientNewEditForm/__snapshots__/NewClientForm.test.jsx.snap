// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Home view accurately 1`] = `
<div>
  <div>
    <h2>
      Client Details
    </h2>
    <fieldset>
      <form>
        <label
          for="client_name"
        >
          Client Name
        </label>
        <input
          id="client_name"
          required=""
          type="text"
          value=""
        />
        <label
          for="email"
        >
          E-Mail
        </label>
        <input
          id="email"
          required=""
          type="email"
          value=""
        />
        <label
          for="phone"
        >
          Phone Number
        </label>
        <input
          id="phone"
          required=""
          type="number"
          value=""
        />
        <label
          for="business_name"
        >
          Business Name
        </label>
        <input
          id="business_name"
          placeholder="Optional"
          type="text"
          value=""
        />
        <label
          for="notes"
        >
          Notes
        </label>
        <textarea
          id="notes"
          placeholder="Optional"
        />
        <button
          type="submit"
        >
          Save
        </button>
      </form>
    </fieldset>
  </div>
</div>
`;
